src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelListItem.test.tsx

```diff
@@ -26,92 +30,177 @@ import {
 
 import { createMainTrackInfo } from 'tests/fixtures/track-panel';
 import { Status } from 'src/shared/types/status';
+import { DrawerView } from 'src/content/app/browser/drawer/drawerState';
 
-describe('<TrackPanelListItem />', () => {
-  beforeEach(() => {
-    jest.resetAllMocks();
-  });
+import * as drawerActions from '../../drawer/drawerActions';
+import * as browserActions from 'src/content/app/browser/browserActions';
+import * as trackPanelActions from 'src/content/app/browser/track-panel/trackPanelActions';
+import { isTrackCollapsed } from 'src/content/app/browser/track-panel/trackPanelSelectors';
+
+jest.mock('../../drawer/drawerActions', () => {
+  const originalModule = jest.requireActual('../../drawer/drawerActions');
 
-  const defaultProps: TrackPanelListItemProps = {
-    categoryName: faker.lorem.words(),
-    trackStatus: Status.SELECTED,
-    defaultTrackStatus: Status.SELECTED,
-    track: createMainTrackInfo(),
-    activeGenomeId: faker.lorem.words(),
-    activeEnsObjectId: faker.lorem.words(),
-    isDrawerOpened: false,
-    drawerView: 'bookmarks',
-    highlightedTrackId: faker.lorem.words(),
-    isCollapsed: false,
+  return {
+    ...originalModule,
     changeDrawerView: jest.fn(),
-    toggleDrawer: jest.fn(),
-    updateTrackStatesAndSave: jest.fn(),
+    setActiveDrawerTrackId: jest.fn()
+  };
+});
+
+jest.mock('src/content/app/browser/browserActions', () => {
+  const originalModule = jest.requireActual(
+    'src/content/app/browser/browserActions'
+  );
+
+  return {
+    ...originalModule,
+    updateTrackStatesAndSave: jest.fn()
+  };
+});
```

Comment:
1) Is there any point in keeping the actual implementations with `jest.requireActual`?
2) If you do not pass the constructor function as a second argument to `jest.mock`, jest should automatically mock all exports from the module (with jest.fn(), I believe). Will it not be sufficient?

Reply:
Thanks! That makes is a lot cleaner.